src/pointcloud/point_cloud.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import struct
+# from .encode_pcd import *
+
+
+class PlanarPointCloud():
+    '''
+    self.X array of shape (n_points, 1)
+    self.Y array of shape (n_points, 1)
+    '''
+    def __init__(self, points:list=[], normalize:bool = False) -> None:
+        # self.points = np.array(points)
+        """_summary_
+
+        Args:
+            points (list, optional): _description_. Defaults to [].
+            normalized (bool, optional): Defines if the input point passed 
+            to the constructor are normalized. Defaults to False.
+            scale (float, optional): None if not normalized.
+        """        
+        
+        self.points = np.array(points)
+        
+        if normalize:
+            self.normalize_point_cloud()
+            
+        self.scale = None #indicates if the pointcloud has been normalized yet
+    
+    @property
+    def X(self):
+        ''' 
+        self.X array of shape (n_points, 1)
+        '''
+        X = self.points[:,0]
+        return X.reshape(-1, 1)
+    
+    @property
+    def Y(self):
+        ''' 
+        self.Y array of shape (n_points, 1)
+        '''
+        Y = self.points[:,1]
+        return Y.reshape(-1, 1)
+       
+    @property 
+    def X_norm(self):
+        if self.scale is None:
```

Comment:
You can normalize and cache the result if someone calls this, rather than creating an exception

Reply:
done